sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1348,10 +1348,12 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   }
 
   long sslOption = 0;
+#if defined(_MSC_VER)
```

Comment:
So, you are first removing the min required libcurl version (7.44), which adds the CURLSSLOPT_NO_REVOKE and then you remove setting that option.
We should try to get libcurl 7.44 for Centos/Redhat instead.  

Removing this option for all non-msvc affects Ubuntu and other POSIX system where libcurl 7.44 is there

Reply:
Yes, the goal here is to make the library compile on a operating system often used in enterprise settings. Right now cmake will refuse to compile because of the minimum version requirement, and the code will fail to compile because of the missing curl option. CURLSSLOPT_NO_REVOKE has only effect on Schannel (the native Windows SSL library). Please revise: https://curl.se/libcurl/c/CURLOPT_SSL_OPTIONS.html

Reply:
So, the request is basically to support libcurl earlier versions than 7.4.
What version are you using?

I will talk and review this with the Azure SDK Core team check if this can be supported or not. At some point in the past we decided not to turn ON/OFF build settings based on libcurl version and instead set the min required version that customers should use.

Can you mention what would it be the reasons why you would not be able to upgrade your libcurl version?

Reply:
These patches are what is needed to be able to compile for RedHat 7. Its an old distribution, but its not EOL. Its often used in HPC environments, but also other enterprise settings where updating the Linux distribution often is not desirable. By shipping our own compiled version of libcurl increases the security audit needed, and introduces unnecessary change into the environment.

Reply:
> These patches are what is needed to be able to compile for RedHat 7.

@jorgen are you using RH v7.0 specifically, or a newer minor version?

> By shipping our own compiled version of libcurl increases the security audit needed, and introduces unnecessary change into the environment.

Do you know what version of libcurl is available on that distro right out of the box? How do you typically install dependencies on such distros?

Reply:
I use a docker image with:
`FROM docker.io/centos:7` so I assume that is the latest CentOS 7 release there is.
CentOS is a binary compatible distribution to RedHat where the packages have the same version and patches applied to them.
```

[root@7b5b85ecf427 /]# yum info libcurl
Loaded plugins: fastestmirror, ovl
Repodata is over 2 weeks old. Install yum-cron? Or run: yum makecache fast
Loading mirror speeds from cached hostfile
 * base: ftpmirror.infania.net
 * centos-sclo-rh: ftpmirror.infania.net
 * centos-sclo-sclo: ftpmirror.infania.net
 * epel: mirrors.glesys.net
 * extras: ftpmirror.infania.net
 * updates: ftpmirror.infania.net
Installed Packages
Name        : libcurl
Arch        : x86_64
Version     : 7.29.0
Release     : 59.el7_9.1
Size        : 429 k
Repo        : installed
From repo   : updates
Summary     : A library for getting files from web servers
URL         : http://curl.haxx.se/
License     : MIT
Description : libcurl is a free and easy-to-use client-side URL transfer library, supporting
            : FTP, FTPS, HTTP, HTTPS, SCP, SFTP, TFTP, TELNET, DICT, LDAP, LDAPS, FILE, IMAP,
            : SMTP, POP3 and RTSP. libcurl supports SSL certificates, HTTP POST, HTTP PUT,
            : FTP uploading, HTTP form based upload, proxies, cookies, user+password
            : authentication (Basic, Digest, NTLM, Negotiate, Kerberos4), file transfer
            : resume, http proxy tunneling and more.

Available Packages
Name        : libcurl
Arch        : i686
Version     : 7.29.0
Release     : 59.el7_9.1
Size        : 226 k
Repo        : updates/7/x86_64
Summary     : A library for getting files from web servers
URL         : http://curl.haxx.se/
License     : MIT
Description : libcurl is a free and easy-to-use client-side URL transfer library, supporting
            : FTP, FTPS, HTTP, HTTPS, SCP, SFTP, TFTP, TELNET, DICT, LDAP, LDAPS, FILE, IMAP,
            : SMTP, POP3 and RTSP. libcurl supports SSL certificates, HTTP POST, HTTP PUT,
            : FTP uploading, HTTP form based upload, proxies, cookies, user+password
            : authentication (Basic, Digest, NTLM, Negotiate, Kerberos4), file transfer
            : resume, http proxy tunneling and more.

[root@7b5b85ecf427 /]# repo
repo-graph   repo-rss     repoclosure  repodiff     repomanage   repoquery    reposync     repotrack
[root@7b5b85ecf427 /]# repoquery -l libcurl
/usr/lib/libcurl.so.4
/usr/lib/libcurl.so.4.3.0
/usr/lib64/libcurl.so.4
/usr/lib64/libcurl.so.4.3.0

```

Reply:
I'm with @jorgen on this one. We should try out best to be compatible with popular Linux distributions, like CentOS 7.

People don't like using `vcpkg` to install dependencies on Linux, so being compatible with default version of libcurl on common Linux distribution is an important step.

Reply:
@jorgen I am curious, what version of cmake does RedHat 7 bring with it? Our SDK requires cmake version `3.13` to build, how are you working with that? Do you manually install or update cmake?

https://github.com/Azure/azure-sdk-for-cpp/blob/1950e6d65524eef42cb9aafa82f6cc86ed36d2c9/sdk/core/CMakeLists.txt#L4

Looks like centOS brings cmake version `2.8.12.2` when doing `yum install`.

Reply:
Yes, I'm downloding a cmake binary from cmake.org and use that. I think its important to make a distinction between tools used to compile a library and userland libraries used by the library.